pkg/services/ngalert/notifier/receiver_svc.go

```diff
@@ -160,6 +177,15 @@ func (rs *ReceiverService) GetReceiver(ctx context.Context, q models.GetReceiver
 // GetReceivers returns a list of receivers a user has access to.
 // Receivers can be filtered by name, and secure settings are decrypted if requested and the user has access to do so.
 func (rs *ReceiverService) GetReceivers(ctx context.Context, q models.GetReceiversQuery, user identity.Requester) ([]*models.Receiver, error) {
+	ctx, span := rs.tracer.Start(ctx, "alerting.receivers.getMany", trace.WithAttributes(
```

Comment:
I'm missing context here so this might have been discussed already, but just curious as to why not starting the traces at the handler level instead? Then we could extract some of this into a middleware

Reply:
Middleware already creates a span but it is quite generic. These spans are enriched with some information that, in my opinion, will help the investigating issues. Also, these spans have some additional events that make operation boundaries a little bit clearer.

Reply:
Ahh, I see. I thought those were supposed to be the root spans 👍